dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TaskDefinitionServiceImpl.java

```diff
@@ -821,7 +821,7 @@ public Map<String, Object> updateTaskWithUpstream(User loginUser, long projectCo
         } else {
             queryUpStreamTaskCodeMap = new HashMap<>();
```

Comment:
I think it's not a bug, because this else branch will create a empty `queryUpStreamTaskCodeMap`.

BTW, this method is difficult to understand and missing comments,  :cry:

Reply:
cc @fuchanghai @ruanwenjun